bluesky/callbacks/best_effort.py

```diff
@@ -441,6 +441,55 @@ def clear(self):
         self._buffer = StringIO()
         self._baseline_toggle = True
 
+    def plot_prune_fifo(self, num_lines, x_signal, y_signal):
+        """
+        Find the plot with axes x_signal and y_signal.  Replot with only the last *num_lines* lines.
+
+        Example to remove all scans but the last:
+        >>> bec.plot_prune_fifo(1, m1, noisy)
+
+        Parameters
+        ----------
+        num_lines: int
+            number of lines (plotted scans) to keep, must be >= 0
+        x_signal: object
+            instance of ophyd.Signal (or subclass), 
+            independent (x) axis
+        y_signal: object
+            instance of ophyd.Signal (or subclass), 
+            dependent (y) axis
+        """
+        if num_lines < 0:
+            emsg = (f"Argument 'num_lines' (given as {num_lines})"
+                    " must be >= 0.")
+            raise ValueError(emsg)
+        for liveplot in self._live_plots.values():
+            lp = liveplot.get(y_signal.name)
+            if lp is None or lp.x != x_signal.name or lp.y != y_signal.name:
+                continue
+
+            # pick out only the lines that contain plot data,
+            # skipping the lines that show peak centers
+            lines = [
+                tr
+                for tr in lp.ax.lines
+                if len(tr._x) != 2 
+                    or len(tr._y) != 2 
+                    or (len(tr._x) == 2 
+                        and tr._x[0] != tr._x[1])
+            ]
+            print(f"found these lines: {lines}")
+            if len(lines) > num_lines:
+                print(f"Limiting LivePlot({y_signal.name}) to {num_lines} lines.")
+                keepers = lines[-num_lines:]
+                print(f"keepers: {keepers}")
+                for tr in lp.ax.lines:
```

Comment:
Should it be looping over `lines` instead of `lp.ax.lines`, as the center lines were removed at this point?

Reply:
I changed it locally, and now it's passing!
```
$ pytest -vvvv -x --pdb bluesky/tests/test_bec.py::test_plot_prune_fifo
================================================= test session starts ==================================================
platform darwin -- Python 3.6.7, pytest-4.5.0, py-1.7.0, pluggy-0.12.0 -- /Users/mrakitin/anaconda3/envs/bluesky/bin/python
cachedir: .pytest_cache
benchmark: 3.2.2 (defaults: timer=time.perf_counter disable_gc=False min_rounds=5 min_time=0.000005 max_time=1.0 calibration_precision=10 warmup=False warmup_iterations=100000)
rootdir: /Users/mrakitin/src/mrakitin/DAMA/bluesky
plugins: faulthandler-1.6.0, benchmark-3.2.2
collected 1 item

bluesky/tests/test_bec.py::test_plot_prune_fifo PASSED                                                           [100%]

=============================================== 1 passed in 4.62 seconds ===============================================
```
I can push my change to your PR branch if you wish.

Reply:
I'm on a roll.  I'll do it from here.

Reply:
I'll do that from here

Reply:
![image](https://user-images.githubusercontent.com/13209176/70279840-3f669c80-1785-11ea-84ed-75bfcfb63983.png)